### DISCUSSION TITLE
Orion Custom GPT

### DISCUSSION THREAD

**Discussion by kmesiab** [kmesiab/orion#35] [General]
### I have been working on fine tuning a custom GPT model: 

https://chat.openai.com/g/g-ECL6eq1jG-code-review-advanced-fine-tuned

The goal is to provide Orion users with a highly specialized code review LLM with a focus on both code quality, and effective code review best practices.

### So far I have trained this model on the following data sets:

**Code Complete, 2nd Edition:** A comprehensive guide that covers a variety of programming topics, focusing on practical techniques for improving the quality of software development. It addresses design, coding, testing, and debugging practices, among others, and emphasizes the importance of readability, maintainability, and efficiency in code.

**OWASP Code Review Guide v2:** This document provides insights into security-focused code review practices, detailing how to identify common security vulnerabilities in code. The OWASP (Open Web Application Security Project) guide is crucial for developers and security analysts aiming to ensure that applications are secure against known threats.

**The Quality of Software by C.A.R. Hoare**: An editorial piece that discusses the essential qualities of good software, including clarity, simplicity, reliability, and efficiency. Hoare emphasizes the need for software to meet a set of comprehensive criteria to be considered high-quality.

**Coding Best Practices**: This document offers guidance on general best practices in coding, which may include conventions on naming, commenting, structuring code, and more, aiming to improve code quality, readability, and maintainability.

[How To Code Review](https://zenodo.org/record/3599150): This document probably focuses on the process of conducting code reviews, offering strategies for effectively reviewing code, identifying issues, and providing constructive feedback to improve software quality and team collaboration.

[Code Review Open Platform](https://zenodo.org/record/3599150): This document is a list of over 100MB worth of highly curated code reviews from large open source projects.

**Comment by frankie-mur** [comment]
Very cool! If I may add a suggestion on a great article I have read recently by Goggle
https://google.github.io/eng-practices/review/reviewer/

**Reply by kmesiab** [reply]
Great!  Adding it shortly!
